release/nightly_tests/dataset/map_batches_benchmark.py

```diff
@@ -0,0 +1,93 @@
+from typing import Literal
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+
+
+def map_batches(
+    input_ds: Dataset,
+    batch_size: int,
+    batch_format: Literal["default", "pandas", "pyarrow", "numpy"],
+    num_calls: int,
+) -> Dataset:
+
+    ds = input_ds
+    for _ in range(num_calls):
+        ds = ds.map_batches(
+            lambda ds: ds, batch_format=batch_format, batch_size=batch_size
+        )
+    return ds
+
+
+def run_map_batches_benchmark(benchmark: Benchmark):
+    input_ds = ray.data.read_parquet(
+        "s3://shuffling-data-loader-benchmarks/data/input_data_0.parquet.snappy"
+    )
+    lazy_input_ds = input_ds.lazy()
+
+    batch_formats = ["pandas", "numpy"]
+    batch_sizes = [64, 128, 256, 512, 1024, 2048, 4096, None]
+    num_calls_list = [1, 2, 4, 8, 16]
+
+    # Test different batch_size of map_batches.
+    for batch_format in batch_formats:
```

Comment:
The other dimension would be different block_format * batch_format combinations, which can reveal conversion cost.

Reply:
do you mean something like

```
ds.map_batches(batch_format="pandas").map_batches(batch_format="numpy")
ds.map_batches(batch_format="numpy").map_batches(batch_format="pandas")
ds.map_batches(batch_format="pandas").map_batches(batch_format="pyarrow")

```

Reply:
Sort of, but the first .map_batches() should be done outside of the benchmark since it's just preparing the dataset format.

Reply:
@jianoaix - sure, added.